### PR TITLE
Issue #3496: Refactor JavadocUtils.containsInBranch method

### PR DESCRIPTION
**Pull request by Vladlis**
#3496 
A separate PR is needed as  method is reimplemented completely.

### CODE REVIEW

**Comment by rnveach** [comment]
@Vladlis 
This looks like the changes could be related to #3496 .
Was there any other reason to redo this logic?

**Comment by Vladlis** [comment]
@rnveach 
Yes, I came across this method while working on #3496, but when I tried a simple solution for `ReturnCount` by replacing `return` with `break` the other violation appeared saying that I cannot use more than one `break` or `continue`, so I decided to reimplement the method completely.
Also the logic of this method was really confusing as for me :)

@rnveach @romani 
If you think that this change does not deserve a separate PR I will submit it as a part of updates for #3496,

**Comment by rnveach** [comment]
My opinion is it shouldn't be it's own standalone commit, especially when PR referenced was not going to give any details on why it was changed (missing first post).
We can keep it part of this PR by itself, just associate it to #3496.

**Comment by Vladlis** [comment]
@rnveach ok, done

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java [hunk 1]
```diff
@@ -332,29 +332,19 @@ public static DetailNode getFirstChild(DetailNode node) {
...
-                toVisit = getNextSibling(curNode);
-                if (toVisit == null) {
-                    curNode = curNode.getParent();
+        boolean result = false;
+        if (type == node.getType()) {
+            result = true;
+        }
+        else {
+            for (DetailNode child : node.getChildren()) {
+                if (containsInBranch(child, type)) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java]: Recursive calling will lead to stack overflow errors if tree happens to become deeply nested.
We had this problem in another area for one of our java grammar tests. I think we should avoid this.

Comment (Vladlis) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java]: ok, let's stay with the existing algorithm

**Pull request closed without merging**